pkg/util/provider/machinecontroller/machine_util_test.go

```diff
@@ -2725,4 +2725,71 @@ var _ = Describe("machine_util", func() {
 			}),
 		)
 	})
+
+	Describe("#Adjust Machine Create Retry Timeout", func() {
+		type setup struct {
+			machine         *machinev1.Machine
+			mcCreateTimeout metav1.Duration
+			createOpRetry   machineutils.RetryPeriod
+		}
+
+		type expect struct {
+			retryAdjusted bool
+		}
+
+		type data struct {
+			setup  setup
+			expect expect
+		}
+
+		DescribeTable("##table",
+			func(data *data) {
+				stop := make(chan struct{})
+				defer close(stop)
+
+				c, trackers := createController(stop, testNamespace, nil, nil, nil, nil)
+				defer trackers.Stop()
+
+				c.safetyOptions.MachineCreationTimeout = data.setup.mcCreateTimeout
+
+				adjustedRetry := c.adjustCreateRetryRequired(data.setup.machine, machineutils.RetryPeriod(data.setup.createOpRetry))
+
+				if data.expect.retryAdjusted {
+					Expect(adjustedRetry).Should(BeNumerically("<=", machineutils.RetryPeriod(c.getEffectiveCreationTimeout(data.setup.machine).Duration)))
```

Comment:
```suggestion
// Considering time.Now() = 0
					Expect(adjustedRetry).Should(BeNumerically("<=", machineutils.RetryPeriod(c.getEffectiveCreationTimeout(data.setup.machine).Duration)))
```
for ease of understanding :)